service/dap/server.go

```diff
@@ -1120,8 +1104,10 @@ func (s *Session) stopDebugSession(killProcess bool) error {
 
 // halt sends a halt request if the debuggee is running.
 // changeStateMu should be held when calling (*Server).halt.
-func (s *Session) halt() (*api.DebuggerState, error) {
-	s.setHaltRequested(true)
+func (s *Session) halt(shouldResume bool) (*api.DebuggerState, error) {
```

Comment:
I think that instead of having a bool argument here, callers should just call setHaltRequested directly when they would pass false to shouldResume.